src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NetFx.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Runtime.Versioning;
+
+namespace System
+{
+    public static partial class PlatformDetection
+    {
+        private static string FrameworkName => AppDomain.CurrentDomain.SetupInformation.TargetFrameworkName;
+        private static Version TargetVersion => new FrameworkName(FrameworkName).Version;
+
+        public static bool IsFullFramework452OrDownTargeting => TargetVersion.CompareTo(new Version(4, 5, 3, 0)) < 0;
```

Comment:
Do we implement these quirks also in CoreFX libraries?
Do we have any test runs (or DirectoryServices special test scripts) which actually run the tests with that TFM set?

What about name `TargetsFullFramework453OrNewer` (or `DoesTargetXYZ`) to distinguish from the "running on" methods and follow the establish pattern XOrNewer/Greater.

Reply:
> Do we implement these quirks also in CoreFX libraries?

No, the old behavior is bad and this is why the full framework fixed it with the quirk. I see no reason doing that.

> Do we have any test runs (or DirectoryServices special test scripts) which actually run the tests with that TFM set?

The xunit runner is actually targeting 4.5.2. so the answer to your question is yes.

> What about name TargetsFullFramework453OrNewer (or DoesTargetXYZ) to distinguish from the "running on" methods and follow the establish pattern XOrNewer/Greater?

That will complicate the check because you need a way to check if it is not a full framework too when using it. using IsFullFramework452OrDownTargeting will allow you to use it without any extra checks

Reply:
> No, the old behavior is bad and this is why the full framework fixed it with the quirk. I see no reason doing that.
> The xunit runner is actually targeting 4.5.2. so the answer to your question is yes.

In that case it feels weird we have 2 code paths in the tests & supporting method for TFM quirks just because there is different behavior on Desktop and one day we may update xunit runner to target 4.5.3+.
I wonder if it wouldn't be easier just hardcode current Desktop behavior (targeting 4.5.2) and update it in future once we update the xunit runner -- we can even leave a comment to hint so. Or just disable the tests on Desktop for now. ... To keep things a bit simpler.
Is it a bad idea?

Reply:
we have the library CoreFx.Private.TestUtilities specifically for such cases. which is the conditions that occur on a platform and not occurring on the others. I am not seeing the current case is different than the other cases we already doing and I am expecting in the future we'll use the current property (or similar properties) to handle other full framework quirk cases. I am preferring the current approach than the idea you have proposed.

Reply:
From offline discussion with @tarekgh: I didn't understand the value of testing the Desktop facades and forwarders - that rules out disabling the test on Desktop.
Having 2 implementations for Desktop in CoreFX still feels a bit of an overkill to me, but it may come handy one day in year or two when we update xunit runner to target higher version, so I don't mind too much.

Maybe adding comment that Targeting comes from xunit runner might be helpful for future readers of the code.